model/SyncService.php

```diff
@@ -525,4 +537,19 @@ private function getSyncId(array $params)
 
         return $params[DataSyncHistoryService::SYNC_NUMBER];
     }
-}
\ No newline at end of file
+
+    private function storeEntityCheckums($entities, array $ids = [])
+    {
+        foreach ($entities as $entity) {
+            if (empty($entity['checksum']) || (!empty($ids) && !in_array($entity['id'], $ids))) {
```

Comment:
Maybe it would be better to have two method: `storeEntityChecksum(array $entities, array $ids)` and `updateEntityChecksum(array $entities)` to avoid complex conditions like this one.

Reply:
I don't think it's too bad, the difference in those 2 methods would be `!empty($ids)` check